ospd_openvas/messaging/mqtt.py

```diff
@@ -96,8 +101,22 @@ def subscribe(
 
         logger.debug("Subscribing to topic %s", message_class.topic)
 
-        self._client.subscribe(message_class.topic, qos=QOS_AT_LEAST_ONCE)
-        self._client.message_callback_add(message_class.topic, func)
+        self.client.subscribe(message_class.topic, qos=QOS_AT_LEAST_ONCE)
+        self.client.message_callback_add(message_class.topic, func)
+
+        self.subscriptions[message_class.topic] = func
+        self.client.user_data_set(self.subscriptions)
```

Comment:
you do not have to set the userdata for each subscription, setting it once in the init method is sufficient